lib/shared/s3-packages.js

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const semver = require('semver');
+const Bluebird = require('bluebird');
+const request = Bluebird.promisifyAll(require('request'));
+const xml = Bluebird.promisifyAll(require('xml2js'));
+
+/**
+ * @summary Etcher production releases S3 bucket URL
+ * @constant
+ * @public
+ * @type {String}
+ */
+exports.BUCKET_URL_PRODUCTION = 'https://resin-production-downloads.s3.amazonaws.com';
+
+/**
+ * @summary Etcher snapshot releases S3 bucket URL
+ * @constant
+ * @public
+ * @type {String}
+ */
+exports.BUCKET_URL_SNAPSHOT = 'https://resin-nightly-downloads.s3.amazonaws.com';
+
+/**
+ * @summary Etcher releases S3 package name
+ * @constant
+ * @private
+ * @type {String}
+ */
+const S3_PACKAGE_NAME = 'etcher';
+
+/**
+ * @summary Number of packages per Etcher version
+ * @constant
+ * @private
+ * @type {Number}
+ */
+const NUMBER_OF_PACKAGES = 8;
+
+/**
+ * @summary Get the correct S3 bucket url from a version
+ * @function
+ * @public
+ *
+ * @param {String} version - application version
+ * @returns {String} S3 bucket url
+ *
+ * @example
+ * const version = require('../../package.json');
+ * const bucketUrl = s3Packages.getBucketUrlFromVersion(version);
+ * console.log(bucketUrl);
+ */
+exports.getBucketUrlFromVersion = (version) => {
+  const GIT_HASH_REGEX = /^[0-9a-f]{7,40}$/;
+
+  if (GIT_HASH_REGEX.test(_.first(semver.parse(version).build))) {
+    return exports.BUCKET_URL_SNAPSHOT;
+  }
```

Comment:
No benefit, I just think that sometimes:
```js
if (a) {
  return 1;
} else {
  return 2;
}
```
looks nicer than
```js
if (a) {
  return 1;
}
return 2;
```

But it doesn't really matter, so feel free to ignore me.